src/Database/LSMTree/Internal/BloomFilter.hs

```diff
@@ -125,3 +129,57 @@ hGetByteArrayExactly hfs h len = do
     _   <- hGetBufExactly hfs h buf 0 (fromIntegral len)
     P.unsafeFreezeByteArray buf
 
+-- Bulk insert
+-----------------------------------------------------------
+
+{-# INLINE bloomInserts #-}
+bloomInserts ::
+     forall a s.
+     BF.Hashable a
+  => BF.MBloom s a
+  -> (Int -> ST s a)
+  -> Int
+  -> ST s ()
+bloomInserts bloom key n =
```

Comment:
I understand how it achieves faster bulk inserts using prefetching because I have seen it before, but maybe a small (internal) note in a comment might make it clear to future readers. Maybe something along the lines of "prefetch blocks of the bit array before inserting into them using a rolling buffer"

Reply:
Yeah.

Reply:
Added a comment on the general strategy.
```
    -- The general strategy is to use a rolling buffer @buf@ (of size 16). At
    -- the write end of the buffer, we prepare the probe locations and prefetch
    -- the corresponding cache line. At the read end, we do the hash insert.
    -- By having a prefetch distance of 15 between the write and read ends, we
    -- can have up to 15 memory reads in flight at once, thus improving
    -- utilisation of the memory bandwidth.
```